app/models/host/managed.rb

```diff
@@ -485,6 +485,11 @@ def host_inherited_params_objects
     params
   end
 
+  def all_parameters
```

Comment:
Nitpick, could you call this `host_params_objects` so it matches the existing pattern?  It's implementing the same as `host_params` on line 464, but with objects rather than hashes.

Perhaps we ought to refactor the non-object methods later to use these as a base, it's beginning to get silly.